src/linear_algebra/matrix/ldl.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2022 Alexander Bentkamp. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alexander Bentkamp
+-/
+import analysis.inner_product_space.gram_schmidt_ortho
+import linear_algebra.matrix.pos_def
+
+/-! # LDL decomposition
+
+This file proves the LDL-decomposition of matricies: Any positive definite matrix `S` can be
+decomposed as `S = LDLᴴ` where `L` is a lower-triangular matrix and `D` is a diagonal matrix.
+
+## Main definitions
+
+ * `LDL.lower` is the lower triangular matrix `L`.
+ * `LDL.lower_inv` is the inverse of the lower triangular matrix `L`.
+ * `LDL.diag` is the diagonal matrix `D`.
+
+## Main result
+
+* `ldl_decomposition` states that any positive definite matrix can be decomposed as `LDLᴴ`.
+
+## TODO
+
+* Prove that `LDL.lower` is lower triangular from `LDL.lower_inv_triangular`.
+
+-/
+
+variables {𝕜 : Type*} [is_R_or_C 𝕜]
+  {n : Type*} [linear_order n] [is_well_order n (<)] [locally_finite_order_bot n]
+
+local notation `⟪`x`, `y`⟫` :=
+  @inner 𝕜 (n → 𝕜) (pi_Lp.inner_product_space (λ _, 𝕜)).to_has_inner x y
+
+open matrix
+open_locale matrix
+variables {S : matrix n n 𝕜} [fintype n] (hS : S.pos_def)
+
+/-- The inverse of the lower triangular matrix `L` of the LDL-decomposition. It is obtained by
+applying Gram-Schmidt-Orthogonalization w.r.t. the inner product induced by `Sᵀ` on the standard
+basis vectors `pi.basis_fun`. -/
+noncomputable def LDL.lower_inv : matrix n n 𝕜 :=
+  @gram_schmidt
+    𝕜 (n → 𝕜) _ (inner_product_space.of_matrix hS.transpose) n _ _ _ (λ k, pi.basis_fun 𝕜 n k)
+
+lemma LDL.lower_inv_gram_schmidt_basis :
```

Comment:
The name suggests something like applying `lower_inv` to `gram_schmidt_basis.to_matrix`, how about:
```suggestion
lemma LDL.lower_inv_eq_gram_schmidt_basis :
```